code/datums/smells/smell.dm

```diff
@@ -0,0 +1,88 @@
+#define SMELL_HORRIBLE 0
+#define SMELL_FOUL 1
+#define SMELL_NEUTRAL 2
+#define SMELL_PLEASANT 3
+#define SMELL_NICE 4
+
+#define SMELL_TRACE 2
+#define SMELL_FAINT 4
+#define SMELL_NORMAL 8
+#define SMELL_INTENSE 16
+
+/datum/smell
+	var/qty
+	var/name
+	var/intensity
+	var/description
+
+/datum/smell/New(var/q)
+	qty=q
+	return src
+
+
+//outputs a string that fully describes a smell
+/mob/living/proc/eval_smell(var/datum/smell/S)
+	if(ishuman(src))
```

Comment:
To implement tiger's suggestion: Make a new proc, that does this check, and only does the parent call if none of the blocking conditions are true. Like so:
```
/mob/living/carbon/human/eval_smell(var/datum/smell/S)
		if((species.flags & NO_BREATHE) || (NO_BREATHE in mutations) || wear_mask)//can't breathe can't smell, masks block smells
		    return
        else
            ..()
```